### PR TITLE
Update cf-1486D.mdx

### PR DESCRIPTION
**Pull request by Keys9873**
Added python and java sols

_If any of the below don't apply to this Pull Request, mark the checkbox as done._

- [_] I have tested my code.
- [_] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [_] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.

### CODE REVIEW

**Review by ryanchou-dev — APPROVED** [review]

File: solutions/silver/cf-1486D.mdx [hunk 1]
```diff
@@ -77,6 +77,100 @@ int main() {
...
 ```
-
 </CPPSection>
+<JavaSection>
+
+```java
+import java.io.*;
+import java.util.*;
+
+public class Main {
```
Comment (SansPapyrus683) [inline solutions/silver/cf-1486D.mdx:87]:
```suggestion
public class MaxMedian {
```

File: solutions/silver/cf-1486D.mdx [hunk 2]
```diff
@@ -77,6 +77,100 @@ int main() {
...
+	public static void main(String[] args) throws Exception {
+		BufferedReader br =
+		    new BufferedReader(new InputStreamReader(System.in));
+		PrintWriter pw = new PrintWriter(System.out);
+		StringTokenizer st = new StringTokenizer(br.readLine());
+		int n = Integer.parseInt(st.nextToken());
+		int k = Integer.parseInt(st.nextToken());
+		st = new StringTokenizer(br.readLine());
```
Comment (SansPapyrus683) [inline solutions/silver/cf-1486D.mdx:95]: space out relevant code segments

File: solutions/silver/cf-1486D.mdx [hunk 3]
```diff
@@ -77,6 +77,100 @@ int main() {
...
+			mn = min(mn, f[i - k])
+		# Update the maximum possible difference.
+		diff = max(diff, f[i] - mn)
+
+	# Update binary search bounds.
+	if diff > 0:
+		lo = median
+	else:
+		hi = median - 1
+print(lo)
```
Comment (SansPapyrus683) [inline solutions/silver/cf-1486D.mdx:173]:
```suggestion

print(lo)
```

File: solutions/silver/cf-1486D.mdx [hunk 4]
```diff
@@ -77,6 +77,100 @@ int main() {
...
+n, k = [int(i) for i in input().split()]
+a = [int(i) for i in input().split()]
+
+# Binary search over the maximum median
+lo = 1
+hi = n
+while lo < hi:
+	median = (lo + hi + 1) // 2
+	# Same as the f described in the editorial
+	f = list()
```
Comment (SansPapyrus683) [inline solutions/silver/cf-1486D.mdx:149]:
```suggestion
	f = []
```

File: solutions/silver/cf-1486D.mdx [hunk 5]
```diff
@@ -77,6 +77,100 @@ int main() {
...
+		int lo = 1;
+		int hi = n;
+		while (lo < hi) {
+			int median = (lo + hi + 1) / 2;
+			// Same as the f described in the editorial
+			int[] f = new int[n];
+			for (int i = 0; i < n; i++) {
+				// If a[i] is greater than or equal to the median, it
+				// contributes 1 to f(i) Otherwise if it less than the median,
+				// it contributes -1 to f(i)
```
Comment (SansPapyrus683) [inline solutions/silver/cf-1486D.mdx:108]: multiline comment

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

**Review by SansPapyrus683 — APPROVED** [review]
will merge once ci is done

**Pull request merged**